### PR TITLE
Updates the Season Converter and Fixes Null Cache Reference -- Copy of #526

### PR DESCRIPTION
**Pull request by UghThatGuyAgain**
It appears @mpgramlich hasn't been active in a while. An explanation of the changes made can be found [here](https://github.com/BenFradet/RiotSharp/pull/526). This is a replica of his forked repo. It doesn't need to be accepted, but because of the inactivity, the latest build seems to have errors (and I know that it prevents me from doing some testing on my project).

### CODE REVIEW

**Comment by UghThatGuyAgain** [comment]
Conflicts should be fixed as well, if someone can review that for me (I didn't do any testing for it, sorry haha).

File: RiotSharp/Endpoints/MatchEndpoint/Enums/Converters/SeasonConverter.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public override bool CanConvert(Type objectType)
                 case 9:
                     return Season.Season2017;
                 case 10:
-                    return Season.PreSeason2018;
+                    return Season.PreSeason2017;
```
Comment (BenFradet) [inline RiotSharp/Endpoints/MatchEndpoint/Enums/Converters/SeasonConverter.cs]: why do we have two preseason 2017 btw?

**Comment by UghThatGuyAgain** [comment]
Must've been my mistake when I fixed the conflicts, sorry about that.

**Comment by BenFradet** [comment]
No worries, I think it's good to go after this change :+1:

File: RiotSharp/RiotApi.cs [hunk 2]
```diff
@@ -114,36 +112,28 @@ private RiotApi(string apiKey, IDictionary<TimeSpan, int> rateLimits)
...
-            Champion = new ChampionEndpoint(rateLimitedRequester);
-            Masteries = new MasteriesEndpoint(rateLimitedRequester);
-            Runes = new RunesEndpoint(rateLimitedRequester);
-            League = new LeagueEndpoint(rateLimitedRequester);
-            Match = new MatchEndpoint(rateLimitedRequester, _cache);
-            Spectator = new SpectatorEndpoint(rateLimitedRequester);
-            ChampionMastery = new ChampionMasteryEndpoint(rateLimitedRequester);
-            ThirdParty = new ThirdPartyEndpoint(rateLimitedRequester);
-            Static = new StaticDataEndpoints(staticEndpointProvider);
+           Summoner = new SummonerEndpoint(rateLimitedRequester, _cache);
```
Comment (JanOuborny) [inline RiotSharp/RiotApi.cs]: Can you indent this properly?

**Comment by UghThatGuyAgain** [comment]
Is that what you were looking for there?

**Comment by JanOuborny** [comment]
Yes, now it's fine.

File: RiotSharp/Endpoints/MatchEndpoint/Enums/Converters/SeasonConverter.cs (hunk 1 above)
Comment (BenFradet) [reply RiotSharp/Endpoints/MatchEndpoint/Enums/Converters/SeasonConverter.cs]: will you be making this change @UghThatGuyAgain ?

**Comment by UghThatGuyAgain** [comment]
Neat, thanks guys! (Two days late, apparently I didn't hit comment :P )

**Comment by BenFradet** [comment]
@UghThatGuyAgain can you fix the preseason thing as per my last comment?

**Comment by UghThatGuyAgain** [comment]
My bad, completely missed that, should be good to go now?

**Pull request merged**